semgrep/semgrep/config_resolver.py

```diff
@@ -144,25 +139,29 @@ def _safe_relative_to(a: Path, b: Path) -> Path:
 
     @staticmethod
     def _convert_config_id_to_prefix(config_id: str) -> str:
+        if ADJUST_FOR_DOCKER:
+            relative_path = SEMGREP_SRC_DIRECTORY
+        else:
+            relative_path = Path.cwd()
+
         at_path = Path(config_id)
-        at_path = Config._safe_relative_to(at_path, Path.cwd())
+        at_path = Config._safe_relative_to(at_path, relative_path)
 
-        prefix = ".".join(at_path.parts[:-1]).lstrip("./").lstrip(".")
+        prefix = ".".join(at_path.parts[:-1]).lstrip("./")
```

Comment:
This 2nd `lstrip` isn't necessary. The first will strip all the characters included in its argument:

```
In [1]: "./.foo".lstrip("./").lstrip(".")                                                               
Out[1]: 'foo'

In [2]: "./.foo".lstrip("./")                                                                           
Out[2]: 'foo'
```